### ISSUE TITLE
Request: Layout change

### ISSUE THREAD

**Issue by Luke-Nukem** [tootsuite/mastodon#5727]
I had a bit of a play with CSS rules in Firefox and came up with this layout;

![screenshot-2017-11-16 mastodon](https://user-images.githubusercontent.com/2015870/32914323-05e60e54-cb7a-11e7-95d6-79dd5d70b9f1.png)

The users test/message box is hidden, and pops up above the columns. The columns have a style added in their html line that is `style="width:calc(100% / column_count)"` which seems to work well - just needs some padding adjustments.

The main reason I experimented with this is that I typically have the browser window open split screen with API ref and such open next to code, occasionally switching to mastodon. When I do switch to mastodon I notice that the layout is rather inefficient in the use of space - eg, if I have only two columns open, they are both very narrow (and so make reading hard), and the text entry area is not used very often so perhaps hiding that + making columns auto width would help?

Note: `width:auto;` doesn't work with the flex columns so the `style="width:calc(100% / column_count)` was required, and needs updating as columns are added/removed/updated.

The tab header bar needs adjusting too, to stop the buttons spacing right out.

What do you think? Worth doing? (I hope so).

**Comment by Cassolotl** [comment]
If it's a CSS thing maybe it could be an option in settings for admins who want to offer that?

**Comment by wxcafe** [comment]
Yeah this looks pretty good for a “compact layout” or a tablet layout imo

**Comment by sophia-park** [comment]
God, we need this. The experience on a tablet is just third rate:
![75861830-7dbf-41e0-b772-bfdc1476b881](https://user-images.githubusercontent.com/20846414/38431927-daea2a68-3993-11e8-9129-12f6b0fd177d.png)

Columns shouldn’t be cut off. The UI should be as well considered as desktop / mobile are.
